test/Spec/BotPlutusInterface/Balance.hs

```diff
@@ -105,3 +165,131 @@ addUtxosForChange = do
   case ebalancedTx of
     Left e -> assertFailure (Text.unpack e)
     Right balancedTx -> txInputs <$> balancedTx @?= Right (Set.fromList [txIn1, txIn2])
+
+dontAddChangeToDatum :: Assertion
+dontAddChangeToDatum = do
+  let scrTxOut' =
+        ScriptChainIndexTxOut
+          valAddr
+          (Right validator) -- (valHash, Just validator)
+          (Right scrDatum) -- (scrDatumHash, Just scrDatum)
+          scrValue
+      scrTxOut = Ledger.toTxOut scrTxOut'
+      usrTxOut' =
+        PublicKeyChainIndexTxOut
+          pkhAddr3
+          (Ada.lovelaceValueOf 1_001_000)
+      usrTxOut = Ledger.toTxOut usrTxOut'
+      -- initState :: MockContractState ()
+      initState =
+        def & utxos .~ [(txOutRef6, scrTxOut), (txOutRef7, usrTxOut)]
+          & contractEnv .~ contractEnv'
+      pabConf :: PABConfig
+      pabConf = def {pcOwnPubKeyHash = pkh3}
+      -- contractEnv' :: ContractEnvironment ()
+      contractEnv' = def {cePABConfig = pabConf}
+
+      -- Input UTxOs:
```

Comment:
Ty for providing additional explanations!

Reply:
Perhaps add a note about your ScriptOutputs are the ones with Datums, and as such, are the ones that shouldn't be modified